File: part2/servers/factories/factory_client.cpp
#include "gen-cpp/Message.h"
#include <thrift/transport/TSocket.h>
#include <thrift/protocol/TJSONProtocol.h>
#include <memory>
#include <iostream>
#include <string>

using namespace apache::thrift::transport;
using namespace apache::thrift::protocol;
using std::make_shared;
using std::shared_ptr;

int main(int argv, char * argc[]) {
    shared_ptr<TTransport> trans;
    trans = make_shared<TSocket>("localhost", 9090);
    trans = make_shared<TFramedTransport>(trans);
    auto proto = make_shared<TJSONProtocol>(trans);
    MessageClient client(proto);

    trans->open();
    std::string msg;
    do {
        client.motd(msg);
        std::cout << msg << std::endl;
        std::cout << "Enter to call motd, 'q' to quit" << std::endl;
        std::getline(std::cin, msg);
    } while (0 != msg.compare("q"));
    trans->close();
}

File: part2/servers/factories/FactoryClient.java

import org.apache.thrift.TException;
import org.apache.thrift.protocol.TProtocol;
import org.apache.thrift.protocol.TJSONProtocol;
import org.apache.thrift.transport.TFramedTransport;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransport;

public class FactoryClient {
  public static void main(String[] args) throws TException {
    TTransport trans = new TFramedTransport(new TSocket("localhost", 8585));
    TProtocol proto = new TJSONProtocol(trans);
    Message.Iface client = new Message.Client(proto);
    trans.open();
    String line;
    do {
       System.out.println("Message from server: " + client.motd());
       System.out.println("Enter to continue, 'q' to quit: ");
       line = System.console().readLine();
    } while (0 != line.compareToIgnoreCase("q"));   
  }
}
